src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,648 @@
+ESC_LOW_FREQ_HZ = 3550.e6
+ESC_HIGH_FREQ_HZ = 3680.e6
+
+# ESC Channel 21 Center Frequency
+ESC_CH21_CF_HZ = 36525.e5
+
+# One Mega Hertz
+MHZ = 1.e6
+
+# Channel bandwidth over which SASs execute the IAP process
+IAPBW_HZ = 5.e6
+
+# GWPZ Area Protection reference bandwidth for the IAP process
+GWPZ_RBW_HZ = 10.e6
+
+# PPA Area Protection reference bandwidth for the IAP process
+PPA_RBW_HZ = 10.e6
+
+# GWPZ and PPA height (m)
+GWPZ_PPA_HEIGHT = 1.5
+
+# In-band insertion loss
+IN_BAND_INSERTION_LOSS = 0.5
+
+# Define an enumeration class named ProtectedEntityType with members
+# 'GWPZ_AREA', 'PPA_AREA', 'FSS_CO_CHANNEL', 'FSS_BLOCKING', 'ESC'
+
+
+class ProtectedEntityType(Enum):
+  GWPZ_AREA = 1
+  PPA_AREA = 2
+  FSS_CO_CHANNEL = 3
+  FSS_BLOCKING = 4
+  ESC = 5
+
+# Global container to store neighborhood distance type of all the protection 
+_DISTANCE_PER_PROTECTION_TYPE = {
+   ProtectedEntityType.GWPZ_AREA :  (GWPZ_NEIGHBORHOOD_DIST, GWPZ_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.PPA_AREA : ( PPA_NEIGHBORHOOD_DIST,  PPA_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_CO_CHANNEL : ( FSS_CO_CHANNEL_NEIGHBORHOOD_DIST,  FSS_CO_CHANNEL_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_BLOCKING : ( FSS_BLOCKING_NEIGHBORHOOD_DIST,  FSS_BLOCKING_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.ESC: (ESC_NEIGHBORHOOD_DIST_A, ESC_NEIGHBORHOOD_DIST_B)
+    }
+
+# Define CBSD grant, i.e., a tuple with named fields of 'latitude',
+# 'longitude', 'height_agl', 'indoor_deployment', 'antenna_azimuth',
+# 'antenna_gain', 'antenna_beamwidth', 'cbsd_category', 
+# 'max_eirp', 'low_frequency', 'high_frequency', 'is_managed_grant'
+CbsdGrantInformation = namedtuple('CbsdGrantInformation',
+                       ['latitude', 'longitude', 'height_agl', 
+                        'indoor_deployment', 'antenna_azimuth', 'antenna_gain',
+                        'antenna_beamwidth', 'cbsd_category', 
+                        'max_eirp', 'low_frequency', 'high_frequency', 
+                        'is_managed_grant'])
+
+# Define protection constraint, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'low_frequency', 'high_frequency'
+ProtectionConstraint = namedtuple('ProtectionConstraint',
+                                  ['latitude', 'longitude', 'low_frequency',
+                                   'high_frequency', 'entity_type'])
+
+# Define FSS Protection Point, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'height_agl', 'max_gain_dbi', 'pointing_azimuth',
+# 'pointing_elevation' 
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                                   ['latitude', 'longitude',
+                                    'height_agl', 'max_gain_dbi',
+                                    'pointing_azimuth', 'pointing_elevation'])
+
+
+# Define ESC information, i.e., a tuple with named fields of
+# 'antenna_height', 'antenna_azimuth', 'antenna_gain', 'antenna_pattern_gain'
+EscInformation = namedtuple('EscInformation',
```

Comment:
Unused in this file.
But I guess these structures supposed to be used by other modules (IAP, ..) ??
Why not latitude/longitude like in the other protected point ?
Note: I might decide later to move the structure in a separate file for being reused by the DPA protection stuff. (nothing to do for now)

Reply:
Yes,   EscInformation is used by IAP and Aggregate Interference modules. As   suggested, not changed anything for now